clinica/iotools/converters/aibl_to_bids/utils/clinical.py

```diff
@@ -334,64 +344,192 @@ def _get_csv_files_for_alternative_exam_date(
+
+    bids_path: Path
+        The path to the BIDS directory.
+
+    Returns
+    -------
+    pd.DataFrame :
+        A pandas DataFrame that contains the scans information for all sessions of all participants.
+    """
+    from clinica.iotools.converter_utils import viscode_to_session
+
+    scans_dict = _init_scans_dict(bids_path)
+
+    study = StudyName.AIBL.value
+
+    scans_specs = _load_specifications(clinical_specifications_folder, "scans.tsv")[
+        [study, f"{study} location", "BIDS CLINICA", "Modalities related"]
+    ].dropna()
+
+    for _, row in scans_specs.iterrows():
+        on_bad_lines = (
+            _flutemeta_badline if "flutemeta" in row[f"{study} location"] else "error"
+        )
+        file = _load_metadata_from_pattern(
+            clinical_data_dir, row[f"{study} location"], on_bad_lines
+        )
+        file["session_id"] = file["VISCODE"].apply(lambda x: viscode_to_session(x))
+
+        for bids_id in scans_dict.keys():
+            original_id = bids_id_factory(StudyName.AIBL)(
+                bids_id
+            ).to_original_study_id()
+            for session in scans_dict[bids_id].keys():
+                try:
+                    value = file[
+                        (file["RID"] == int(original_id))
+                        & (file["session_id"] == session)
+                    ][row[study]].item()
+                    if value == "-4":
+                        value = "n/a"
+                    elif row["BIDS CLINICA"] == "acq_time":
+                        value = _format_time(value)
+                except ValueError:
+                    value = "n/a"
+
+                modality = scans_dict[bids_id][session][row["Modalities related"]]
+
+                # Avoid writing over in case of modality "T1/..." because it is used twice
+                if (row["BIDS CLINICA"] not in modality) or (
+                    modality[row["BIDS CLINICA"]] == "n/a"
+                ):
+                    modality[row["BIDS CLINICA"]] = value
+    return scans_dict
+
+
+def write_scans_tsv(bids_dir: Path, scans_dict: dict) -> None:
+    """Write the scans dict into TSV files.
+
+    Parameters
+    ----------
+    bids_dir : Path
+        The path to the BIDS directory.
+
+    scans_dict : dict
+        Dictionary containing scans metadata.
+
+        .. note::
+            This is the output of the function
+            `clinica.iotools.bids_utils.create_scans_dict`.
+
+    See also
+    --------
+    write_sessions_tsv
+    """
+
+    from clinica.iotools.bids_utils import _get_pet_tracer_from_filename
+
+    supported_modalities = ("anat", "dwi", "func", "pet")
+
+    for sub in scans_dict.keys():
+        for session in scans_dict[sub].keys():
+            scans_df = pd.DataFrame()
+            tsv_file = bids_dir / sub / session / f"{sub}_{session}_scans.tsv"
+            tsv_file.unlink(missing_ok=True)
+            for mod in (bids_dir / sub / session).glob("*"):
```

Comment:
Same idea about names and simplify by replacing the following if condition by a list filter:

```suggestion
            for modality in [mod for mod in (bids_dir / subject / session).glob("*") if mod.name in supported_modalities]:
```

Reply:
Much better indeed